<!--
  ~ Copyright 2015-2016 OpenCB
  ~
  ~ Licensed under the Apache License, Version 2.0 (the "License");
  ~ you may not use this file except in compliance with the License.
  ~ You may obtain a copy of the License at
  ~
  ~     http://www.apache.org/licenses/LICENSE-2.0
  ~
  ~ Unless required by applicable law or agreed to in writing, software
  ~ distributed under the License is distributed on an "AS IS" BASIS,
  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  ~ See the License for the specific language governing permissions and
  ~ limitations under the License.
  -->

<dom-module id="faq-web">
    <template>
        <!--<style include="jso-styles">-->
        <style>
            .faq-center {
                margin: auto;
                width: 65%;
                text-align: justify;
                font-size: 16px;
                color: #797979;
            }
        </style>

        <!-- This is where main application is rendered -->
        <div class="faq-center">
            <h1>Frequently Asked Questions</h1>
            <hr>

            <h2>General</h2>

            <h3>How should I cite discoveries made using the rare disease data set?</h3>
            <p>We request that any use of data obtained from the Rare Disease genomes browser cite the bioRxiv
                <a href="https://www.biorxiv.org/content/10.1101/507244v1" target="_blank">paper</a>.
                This will be updated when a peer reviewed version is published.</p>
            <p>There is no need to include us as authors on your manuscript, unless we contributed specific advice or analysis for your work.
                However, we ask that the Consortium be acknowledged in publications as follows:</p>
            <p>“This research was made possible through access to the data and findings generated by two pilot studies for the 100,000 Genomes Project.
                The enrolment for one pilot study was coordinated by the NIHR BioResource and the other by Genomics England Limited (GEL), a wholly owned company of the Department of Health in the UK. These studies were mainly funded by grants from the National Institute for Health Research (NIHR) in England, with additional funding provided by the BHF, MRC, NHS England, the Wellcome Trust and many other fund providers. The studies use data provided by patients and their close relatives collected by the NHS and other healthcare providers as part of their care and support. The vast majority of participants have been enrolled in the NIHR BioResource. We thank all volunteers for their participation, and also gratefully acknowledge NIHR Biomedical Research Centres, NIHR BioResource Centres, NHS Trust Hospitals, NHS Blood and Transplant and staff for their contribution. This research has been conducted using the UK Biobank Resource under Application Number 9616, granting access to DNA samples and accompanying participant data. UK Biobank has received funding from the MRC, Wellcome Trust, Department of Health, British Heart Foundation (BHF), Diabetes UK, Northwest Regional Development Agency, Scottish Government, and Welsh Assembly Government. The MRC and Wellcome Trust played a key role in the decision to establish UK Biobank.”</p>

            <h3>Can I get access to individual-level genotype data?</h3>
            It is not possible to access the individual-level phenotypic data through the allele frequency browser.
            See Data availability section in the <a href="https://www.biorxiv.org/content/10.1101/507244v1" target="_blank">paper</a>
            for further details on genotype, phenotype and clinical interpretation accessions.

            <h3>What are the restrictions on data usage?</h3>
            Allele frequency variant data is available through this web browser and its use should be acknowledged in any publications (see above).
            Individual level data will be available, for specific application mechanisms see Data availability section of the
            <a href="https://www.biorxiv.org/content/10.1101/507244v1" target="_blank">paper</a>.


            <h2 style="padding-top: 20px">Technical Details</h2>

            <h3>What genome build is the data based on?</h3>
            <p>Data are available based on GRCh37/hg19 (Isaac calls) and GRCh38/hg38 (GENALICE), technical details are provided in
                <a href="https://www.biorxiv.org/content/10.1101/507244v1.supplementary-material" target="_blank">supplementary information</a>.</p>

            <h3>What populations are represented in this dataset?</h3>
            <p>The genomic ancestry was determined with a principal component analysis see <a href="https://www.biorxiv.org/content/10.1101/507244v1.supplementary-material" target="_blank">supplementary information</a></p>
            <p>Ancestry breakdown:</p>
            <div style="padding-left: 50px">
                Non-Finnish European (NFE): 82%<br>
                South Asian (SAS): 7.2%<br>
                African (AFR): 2.1%<br>
                East Asian (EAS): 0.7%<br>
                Finnish (FIN): 0.1%<br>
                Other / mixed ethnicity: 7.9%<br>
            </div>
            <h3>What cohorts are represented in this dataset?</h3>
            <p>The dataset contains patients with rare disorders as well as their healthy relatives</p>
            <div style="padding-left: 50px">
                Affected individuals: 75.2%<br>
                Unaffected individuals/ Not known (late onset conditions): 24.8%<br>
            </div>
            <p style="padding-top: 20px">Rare disorders:</p>
            <div style="padding-left: 50px;padding-bottom: 20px">
                Bleeding, Thrombotic and Platelet Disorders<br>
                Cerebral Small Vessel Disease<br>
                Ehler-Danlos and Ehler-Danlos-like Syndromes<br>
                100,000 Genomes Project–Rare Diseases Pilot<br>
                Hypertrophic Cardiomyopathy<br>
                Intrahepatic Cholestasis of Pregnancy<br>
                Inherited Retinal Disorders<br>
                Leber Hereditary Optic Neuropathy<br>
                Multiple Primary Malignant Tumours<br>
                Neurological and Developmental Disorders<br>
                Neuropathic Pain Disorders<br>
                Pulmonary Arterial Hypertension<br>
                Primary Immune Disorders<br>
                Primary Membranoproliferative Glomerulonephritis<br>
                Stem cell and Myeloid Disorders<br>
                Steroid Resistant Nephrotic Syndrome<br>
                UK Biobank –Extreme Red Cell Traits<br>
            </div>
            <p>For a full description of the rare disease domains see
                <a href="https://www.biorxiv.org/content/10.1101/507244v1.supplementary-material" target="_blank">supplementary information</a></p>

        </div>
    </template>

    <script>

        class Faq extends Polymer.Element {

            constructor() {
                super();
            }

            static get is() {
                return 'faq-web';
            }

            static get properties() {
                return {
                    version: {
                        type: String
                    }
                }
            }

            _attachDom(dom) {
                this.appendChild(dom);
            }

        }

        customElements.define(Faq.is, Faq);
    </script>
</dom-module>
