<!--
  ~ Copyright 2015-2016 OpenCB
  ~
  ~ Licensed under the Apache License, Version 2.0 (the "License");
  ~ you may not use this file except in compliance with the License.
  ~ You may obtain a copy of the License at
  ~
  ~     http://www.apache.org/licenses/LICENSE-2.0
  ~
  ~ Unless required by applicable law or agreed to in writing, software
  ~ distributed under the License is distributed on an "AS IS" BASIS,
  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  ~ See the License for the specific language governing permissions and
  ~ limitations under the License.
  -->

<dom-module id="welcome-web">
    <template>
        <!--<style include="jso-styles">-->
        <style>
            .welcome-center {
                margin: auto;
                width: 80%;
                text-align: justify;
                font-size: 18px;
                color: #797979;
            }
        </style>

        <!-- This is where main application is rendered -->
        <div class="welcome-center">
            <div style="text-align: center">
                <h1 style="padding: 50px 0px">Genomics England Web Data Portal (Beta)</h1>
                <div style="font-size: 14px; text-align: justify;padding: 25px 0px">
                    <img src="img/Genomics-England-logo_colour.jpg" width="160px" style="float: left;margin: 0px 20px 20px 0px">

                    <p>
                        The Interactive Variant Analysis (IVA) tool enables whole genome variant browsing and analysis of the Genomics England dataset.
                    </p>

                    <p>
                        This interactive tool provides the functionality to query genetic variants across the cohort and allows you to find
                        genes affected by deleterious variants that segregate along family pedigrees, case-control, or sporadic samples.
                    </p>

                    <p>
                        The Variant Browser tab allows you to efficiently browse annotated and indexed variant studies.
                        You can query variants using various filters such as: population frequency, consequence type, and phenotype associations.
                    </p>

                    <p>
                        The Genome Browser tab allows for an interactive and smooth variant visualisation.
                    </p>

                    <p>
                        The IVA database currently holds all interpreted genomes that are available from the 100,000 Genomes Project from Genomics England.
                        These are divided into the following studies: rare disease GRCh37 and GRCh38 (RD37/RD38), cancer germline GRCh38 (CG38), and cancer somatic GRCh38 (CS38).
                    </p>

                    <div style="padding: 20px 10px">
                        <input id="welcomeSearchTextBox" type="text" class="form-control input-lg" list="FeatureDatalist" on-keyup="callAutocomplete"
                               placeholder="Search for gene symbols, genomic regions or variants" value="">
                        <datalist id="FeatureDatalist"></datalist>
                        <!-- Examples -->
                        <span style="font-size: 0.8em; padding-left: 10px">
                    Examples - Gene: <a on-click="onExampleClick" data-type="gene" style="cursor: pointer">BRCA2</a>,
                    Region: <a on-click="onExampleClick" data-type="region" style="cursor: pointer">3</a>, <a on-click="onExampleClick" data-type="region" style="cursor: pointer">3:113000-1150000</a>,
                    SNP: <a on-click="onExampleClick" data-type="snp" style="cursor: pointer">rs445909</a>
                    Variant: <a on-click="onExampleClick" data-type="variant" style="cursor: pointer">13:32962274:G:T</a>
                </span>
                    </div>

                </div>
            </div>

            <div class="col-sm-12">
                <br><br>
                <span style="font-size:12px">
                    Note:
                    <br> GEL web application makes an intensive use of the HTML5 standard and other cutting-edge web technologies such as Web Components, so only modern web browsers are fully supported.
                </span>
            </div>
        </div>
    </template>

    <script>

        class Welcome extends Polymer.Element {

            constructor() {
                super();
            }

            static get is() {
                return 'welcome-web';
            }

            static get properties() {
                return {
                    version: {
                        type: String
                    },
                    cellbaseClient: {
                        type: Object
                    }
                }
            }

            _attachDom(dom) {
                this.appendChild(dom);
            }

            // onBlur(e) {
            //     this.notify();
            // }

            // onKeyup(e) {
            //     if (e.keyCode === 13) {
            //         this.notify();
            //     }
            // }

            onExampleClick(e) {
                let query = {};
                switch (e.currentTarget.dataset.type) {
                    case "gene":
                        query.gene = e.currentTarget.text;
                        break;
                    case "region":
                        query.region = e.currentTarget.text;
                        break;
                    case "snp":
                        query.xref = e.currentTarget.text;
                        break;
                    case "variant":
                        query.xref = e.currentTarget.text;
                        break;
                }
                this.notify(query);
            }

            notify(query) {
                this.dispatchEvent(new CustomEvent("search", {
                    detail: query,
                    bubbles: true,
                    composed: true
                }));
            }

            callAutocomplete(e) {
                // Only gene symbols are going to be searched and not Ensembl IDs
                let featureId = PolymerUtils.getElementById("welcomeSearchTextBox").value;
                if (UtilsNew.isNotEmpty(featureId)) {
                    let query = {};
                    if (featureId.startsWith("chr") || featureId.match(/^\d/)) {
                        if (featureId.split(":").length < 3) {
                            // It's a region, contains only one ':' character
                            query.region = featureId;
                        } else {
                            query.xref = featureId;
                        }
                    } else if (featureId.startsWith("rs")) {
                        query.xref = featureId;
                    } else {
                        // The ID written seems to be a gene name
                        query.gene = featureId;
                        if (featureId.length >= 3 && !featureId.startsWith("ENS")) {
                            let _this = this;
                            _this.cellbaseClient.get('feature', 'id', featureId.toUpperCase(), 'starts_with', {}, {})
                                .then(function (response) {
                                    let options = "";
                                    for (let id of response.response[0].result) {
                                        options += `<option value="${id.name}">`;
                                    }
                                    PolymerUtils.innerHTML("FeatureDatalist", options);
                                });
                        }
                    }

                    if (e.keyCode === 13) {
                        this.notify(query);
                        PolymerUtils.getElementById("welcomeSearchTextBox").value = "";
                    }

                } else {
                    PolymerUtils.innerHTML("FeatureDatalist", "");
                }
            }
        }

        customElements.define(Welcome.is, Welcome);
    </script>
</dom-module>
