<!--
  ~ Copyright 2015-2016 OpenCB
  ~
  ~ Licensed under the Apache License, Version 2.0 (the "License");
  ~ you may not use this file except in compliance with the License.
  ~ You may obtain a copy of the License at
  ~
  ~     http://www.apache.org/licenses/LICENSE-2.0
  ~
  ~ Unless required by applicable law or agreed to in writing, software
  ~ distributed under the License is distributed on an "AS IS" BASIS,
  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  ~ See the License for the specific language governing permissions and
  ~ limitations under the License.
  -->

<link rel="import" href="../bower_components/polymer/polymer.html">

<link rel="import" href="jso-styles.html">

<dom-module id="welcome-web">

    <template>
        <style is="custom-style" include="jso-styles"></style>
        <style>
            .center {
                margin: auto;
                text-align: justify;
                font-size: 18px;
                color: #797979;
            }
        </style>

        <!-- This is where main application is rendered -->
        <div class="center">
            <br>
            <span style="font-size:50px">The Human Genetic Variation Archive 1.0.0 (HGVA)</span>
            <h2>Overview</h2>
            <p>
                Quite recently NGS techniques started to be used for the identification of trait-causing genomics
                variants. Interestingly, a sheer wealth of candidate causal mutations can be found in any human genome
                (~5 million per human genome, ~150.000 per human exome) many of which may provide a compelling story
                about how the variant may influence the trait; the so-called narrative potential of human genomes. The
                first step for these data to be translated into effective knowledge (e.g. clinical knowledge) is
                through the integration of reference annotation data that enables their filtering and provides a
                context for accurate interpretation.
                <br><br>
                The ability to obtain the list of genomic variants in any human genome has paved the way for the
                development of massive reference projects such as "The 1000 Genomes Project", which has sequenced the
                genome of more than 3000 individuals, The NHLBI Exome Sequencing Project, which has sequenced more than
                6500 exomes or "The Genome of the Netherlands" who sequenced genomes from more than 700 individuals,
                among others. These projects are generating a genomic variation knowledgebase which sets an essential
                base to identifying disease-causing genotypes.
                <br><br>
                <b>The Human Genomic Variation Archive (HGVA)</b> is an open access genetic variation resource that integrates
                all variants from key world-wide reference projects, but also added-value information such as basic
                variant annotation, population frequencies, protein effect predictions, variant-associated phenotyes,
                etc. HGVA currently hosts about 300GB of data from 13 different studies describing more than 200
                million variants. HGVA is not a mere data archive, but a big data provider that enables users to
                efficiently query, filter and retrieve relevant information from its knowledge-base, either from a
                visual web-interface or programatically.
            </p>
            <br>
            <h4>Note:</h4>
            <p style="font-size:14px">
                HGVA web application makes an intensive use of the HTML5 standard and other cutting-edge web technologies such as Web Components,
                so only modern web browsers are fully supported, these include Chrome 49+, Firefox 45+, Microsoft Edge 14+, Safari 10+ and Opera 36+.
                </p>
            <br>
            <br>
            <br>
            <img src="images/opencb-logo.png" width="220">
        </div>
    </template>

    <script>
        Polymer({
            is: 'welcome-web',
            properties: {
                opencgaClient: {
                    type: Object
                },
                host: {
                    type: String,
                    value: "http://bioinfodev.hpc.cam.ac.uk/hgva"
                },
                userId: {
                    type: String,
                    value: ""
                },
                project: {
                    type: String,
                    value: ""
                },
                study: {
                    type: Object
                }
            }
        });
    </script>
</dom-module>
